src/backend/distributed/connection/placement_connection.c

```diff
@@ -846,6 +846,12 @@ ConnectionModifiedPlacement(MultiConnection *connection)
 		return false;
 	}
 
+	if (connection->remoteTransaction.transactionFailed)
```

Comment:
@marcocitus Do you see any potential problem with this change? I am asking since this is a critical method

Reply:
No, I think we mainly use this to check whether we should send a prepared transaction.

There's one additional call from Activate2PCIfModifyingTransactionExpandsToNewNode, but we should not get there for failed connections.

Reply:
Minor: I think the caller of `ConnectionModifiedPlacement` should not call this function if the connection failed. It feels confusing when reading the code. I don't want future callers to assume that failed transactions are OK to check  `ConnectionModifiedPlacement`

In the first iteration of `CoordinatedRemoteTransactionsPrepare` we already skip failed transactions.

I think we should not change this function and make the assert cover failed transactions such as:
```C
/* have some comment*/
Assert( transaction->transactionFailed || !ConnectionModifiedPlacement(connection));
```